aws/templates/solution/solution_userpool.ftl

```diff
@@ -190,125 +173,189 @@
+                    )
+                ]
+                federatedServices="cognito-identity.amazonaws.com"
+                condition={
+                    "StringEquals": {
+                        "cognito-identity.amazonaws.com:aud": getReference(identityPoolId)
+                    },
+                    "ForAnyValue:StringLike": {
+                        "cognito-identity.amazonaws.com:amr": "unauthenticated"
+                    }
                 }
-            }
-        /]
-
-        [@createRole 
-            mode=listMode
-            id=identityPoolAuthRoleId
-            policies=[
-                getPolicyDocument(
-                    getUserPoolAuthPolicy(),
-                    "DefaultAuthIdentityRole"
-                )
-            ]
-            federatedServices="cognito-identity.amazonaws.com"
-            condition={
-                "StringEquals": {
-                    "cognito-identity.amazonaws.com:aud": getReference(identityPoolId)
-                },
-                "ForAnyValue:StringLike": {
-                    "cognito-identity.amazonaws.com:amr": "authenticated"
+            /]
+
+            [@createRole 
+                mode=listMode
+                id=identityPoolAuthRoleId
+                policies=[
+                    getPolicyDocument(
+                        getUserPoolAuthPolicy(),
+                        "DefaultAuthIdentityRole"
+                    )
+                ]
+                federatedServices="cognito-identity.amazonaws.com"
+                condition={
+                    "StringEquals": {
+                        "cognito-identity.amazonaws.com:aud": getReference(identityPoolId)
+                    },
+                    "ForAnyValue:StringLike": {
+                        "cognito-identity.amazonaws.com:amr": "authenticated"
+                    }
                 }
-            }
-        /]
-
-        [@createIdentityPoolRoleMapping
-            mode=listMode
-            component=component
-            tier=tier
-            id=identityPoolRoleMappingId
-            identityPoolId=getReference(identityPoolId)
-            authenticatedRoleArn=getReference(identityPoolAuthRoleId, ARN_ATTRIBUTE_TYPE)
-            unauthenticatedRoleArn=getReference(identityPoolUnAuthRoleId, ARN_ATTRIBUTE_TYPE)
-        /]
+            /]
+
+            [@createIdentityPoolRoleMapping
+                mode=listMode
+                component=component
+                tier=tier
+                id=identityPoolRoleMappingId
+                identityPoolId=getReference(identityPoolId)
+                authenticatedRoleArn=getReference(identityPoolAuthRoleId, ARN_ATTRIBUTE_TYPE)
+                unauthenticatedRoleArn=getReference(identityPoolUnAuthRoleId, ARN_ATTRIBUTE_TYPE)
+            /]
+        [/#if]
+        
+        [#if deploymentSubsetRequired("epilogue", false)]
+            [@cfScript
+                mode=listMode
+                content=
+                [] +
```

Comment:
Is this empty array needed? The following function returns an empty array if there isn't any content